### PR TITLE
Moved database-plugins code to database-commons

### PR DESCRIPTION
**Pull request by ibarbashov**
Except database-plugins code, this PR contains maven configuration to handle every specific database as a different submodule.

### CODE REVIEW

File: database-commons/src/main/java/co/cask/DBManager.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2015-2018 Cask Data, Inc.
```
Comment (albertshau) [inline database-commons/src/main/java/co/cask/DBManager.java]: please update all copyrights to 2019 since these are now new files.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,882 @@
...
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-api-spark</artifactId>
+        <version>${cdap.version}</version>
+        <scope>provided</scope>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-etl-proto</artifactId>
+        <version>${cdap.version}</version>
+        <scope>provided</scope>
```
Comment (albertshau) [inline pom.xml]: I believe nothing from proto should be used in the plugin source code itself, only in the unit tests. If so, this should be test scope

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,882 @@
...
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-api</artifactId>
+        <version>${cdap.version}</version>
+        <scope>provided</scope>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-api-spark</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,882 @@
...
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-etl-api</artifactId>
+        <version>${cdap.version}</version>
+        <scope>provided</scope>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-etl-api-spark</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${cdap.version}</version>
+        <scope>provided</scope>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-formats</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,882 @@
...
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-data-pipeline</artifactId>
+        <version>${cdap.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-data-streams</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,882 @@
...
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.hydrator</groupId>
+        <artifactId>hydrator-common</artifactId>
+        <version>${hydrator.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.cdap</groupId>
+        <artifactId>cdap-etl-realtime</artifactId>
```
Comment (albertshau) [inline pom.xml]: this is not needed

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${guava.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>commons-lang</groupId>
+        <artifactId>commons-lang</artifactId>
+        <version>${commons-lang-version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.hadoop</groupId>
+        <artifactId>hadoop-minicluster</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,882 @@
...
+      </dependency>
+      <dependency>
+        <groupId>junit</groupId>
+        <artifactId>junit</artifactId>
+        <version>${junit.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.cassandra</groupId>
+        <artifactId>cassandra-all</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${cassandra.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.xerial.snappy</groupId>
+        <artifactId>snappy-java</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${snappy.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.cassandraunit</groupId>
+        <artifactId>cassandra-unit</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${cassandraunit.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.datastax.cassandra</groupId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,882 @@
...
+        <artifactId>cassandra-driver-core</artifactId>
+        <version>${datastax.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.twitter4j</groupId>
+        <artifactId>twitter4j-core</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 14]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${twitter4j.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.twitter4j</groupId>
+        <artifactId>twitter4j-stream</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 15]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${twitter4j.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.mockftpserver</groupId>
+        <artifactId>MockFtpServer</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 16]
```diff
@@ -0,0 +1,882 @@
...
+      </dependency>
+      <dependency>
+        <groupId>org.hsqldb</groupId>
+        <artifactId>hsqldb</artifactId>
+        <version>${hsql.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.elasticsearch</groupId>
+        <artifactId>elasticsearch</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 17]
```diff
@@ -0,0 +1,882 @@
...
+          </exclusion>
+          <exclusion>
+            <groupId>javax.servlet</groupId>
+            <artifactId>servlet-api</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.elasticsearch</groupId>
+        <artifactId>elasticsearch-hadoop-mr</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 18]
```diff
@@ -0,0 +1,882 @@
...
+        <exclusions>
+          <exclusion>
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.avro</groupId>
+        <artifactId>avro</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 19]
```diff
@@ -0,0 +1,882 @@
...
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.elasticsearch</groupId>
+        <artifactId>elasticsearch-hadoop-mr</artifactId>
+        <version>${es-hadoop.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>commons-httpclient</groupId>
+        <artifactId>commons-httpclient</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 20]
```diff
@@ -0,0 +1,882 @@
...
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.avro</groupId>
+        <artifactId>avro</artifactId>
+        <version>${avro.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.avro</groupId>
+        <artifactId>avro-mapred</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 21]
```diff
@@ -0,0 +1,882 @@
...
+          </exclusion>
+          <exclusion>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.geronimo.specs</groupId>
+        <artifactId>geronimo-jms_1.1_spec</artifactId>
```
Comment (albertshau) [inline pom.xml]: don't think this is needed

File: pom.xml [hunk 22]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${geronimo-jms.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.zookeeper</groupId>
+        <artifactId>zookeeper</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 23]
```diff
@@ -0,0 +1,882 @@
...
+          </exclusion>
+          <exclusion>
+            <groupId>jline</groupId>
+            <artifactId>jline</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.twill</groupId>
+        <artifactId>twill-core</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 24]
```diff
@@ -0,0 +1,882 @@
...
+        <exclusions>
+          <exclusion>
+            <groupId>org.scala-lang</groupId>
+            <artifactId>scala-compiler</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.kafka</groupId>
+        <artifactId>kafka_2.10</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 25]
```diff
@@ -0,0 +1,882 @@
...
+          </exclusion>
+          <exclusion>
+            <groupId>org.scala-lang</groupId>
+            <artifactId>scala-compiler</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.ow2.asm</groupId>
+        <artifactId>asm-all</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 26]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${asm.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.python</groupId>
+        <artifactId>jython-standalone</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 27]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${jython.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>dumbster</groupId>
+        <artifactId>dumbster</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 28]
```diff
@@ -0,0 +1,882 @@
...
+          </exclusion>
+          <exclusion>
+            <groupId>javax.activation</groupId>
+            <artifactId>activation</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>javax.mail</groupId>
+        <artifactId>mail</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 29]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${javamail.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.spark</groupId>
+        <artifactId>spark-streaming-twitter_2.10</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 30]
```diff
@@ -0,0 +1,882 @@
...
+        <version>1.6.2</version>
+      </dependency>
+      <dependency>
+        <groupId>io.netty</groupId>
+        <artifactId>netty-all</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

File: pom.xml [hunk 31]
```diff
@@ -0,0 +1,882 @@
...
+        <version>${netty.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>co.cask.http</groupId>
+        <artifactId>netty-http</artifactId>
```
Comment (albertshau) [inline pom.xml]: not needed

**Comment by albertshau** [comment]
there are still a lot of dependencies that shouldn't be needed. Elastic search, avro, etc.

File: database-commons/src/main/java/co/cask/DBManager.java (hunk 1 above)
Comment (ibarbashov) [reply database-commons/src/main/java/co/cask/DBManager.java]: done

File: pom.xml (hunk 2 above)
Comment (ibarbashov) [reply pom.xml]: done

File: pom.xml (hunk 3 above)
Comment (ibarbashov) [reply pom.xml]: removed

File: pom.xml (hunk 4 above)
Comment (ibarbashov) [reply pom.xml]: removed

File: pom.xml (hunk 5 above)
Comment (ibarbashov) [reply pom.xml]: removed

File: pom.xml (hunk 27 above)
Comment (ibarbashov) [reply pom.xml]: removed

File: pom.xml (hunk 28 above)
Comment (ibarbashov) [reply pom.xml]: removed

File: pom.xml (hunk 29 above)
Comment (ibarbashov) [reply pom.xml]: removed

File: pom.xml (hunk 30 above)
Comment (ibarbashov) [reply pom.xml]: removed

File: pom.xml (hunk 31 above)
Comment (ibarbashov) [reply pom.xml]: removed

**Comment by albertshau** [comment]
There are still two dependencies left that I commented on (cdap-etl-realtime and cdap-data-streams). etl-realtime for sure is not needed. data-streams is only needed if streaming pipelines are being unit tested, which doesn't seem to be the case.

Please also remove the properties that are no longer used.

**Pull request merged**